src/oauth-shim.js

```diff
@@ -196,12 +196,34 @@ oauth_shim.redirect = function(req, res, next) {
 
 		path += (hash ? '#' + param(hash) : '');
 
-		res.writeHead(302, {
-			'Access-Control-Allow-Origin': '*',
-			'Location': path
-		});
+		try {
+			res.writeHead(302, {
+				'Access-Control-Allow-Origin': '*',
+				'Location': path
+			});
+
+			res.end();
+		} catch(err) {
+			res.writeHead(500, {
```

Comment:
should be 502

Reply:
As discussed, we are going to assume that if we've reached this point, it's definitely the authenticating server's fault for not formatting the data correctly (to spec)